src/server/android/android.ts

```diff
@@ -51,35 +53,64 @@ export interface SocketBackend extends EventEmitter {
 
 export class Android {
   private _backend: Backend;
+  readonly _timeoutSettings: TimeoutSettings;
 
   constructor(backend: Backend) {
     this._backend = backend;
+    this._timeoutSettings = new TimeoutSettings();
+  }
+
+  setDefaultTimeout(timeout: number) {
+    this._timeoutSettings.setDefaultTimeout(timeout);
   }
 
   async devices(): Promise<AndroidDevice[]> {
     const devices = await this._backend.devices();
-    return await Promise.all(devices.map(d => AndroidDevice.create(d)));
+    return await Promise.all(devices.map(d => AndroidDevice.create(this, d)));
   }
 }
 
-export class AndroidDevice {
+export class AndroidDevice extends EventEmitter {
   readonly _backend: DeviceBackend;
   readonly model: string;
   readonly serial: string;
   private _driverPromise: Promise<Transport> | undefined;
   private _lastId = 0;
   private _callbacks = new Map<number, { fulfill: (result: any) => void, reject: (error: Error) => void }>();
+  private _pollingWebViews: NodeJS.Timeout | undefined;
+  readonly _timeoutSettings: TimeoutSettings;
+  private _webViews = new Map<number, AndroidWebView>();
+
+  static Events = {
+    WebViewAdded: 'webViewAdded',
+    WebViewRemoved: 'webViewRemoved',
+  };
+
+  private _browserConnections = new Set<AndroidBrowser>();
 
-  constructor(backend: DeviceBackend, model: string) {
+  constructor(android: Android, backend: DeviceBackend, model: string) {
+    super();
     this._backend = backend;
     this.model = model;
     this.serial = backend.serial;
+    this._timeoutSettings = new TimeoutSettings(android._timeoutSettings);
   }
 
-  static async create(backend: DeviceBackend): Promise<AndroidDevice> {
+  static async create(android: Android, backend: DeviceBackend): Promise<AndroidDevice> {
     await backend.init();
     const model = await backend.runCommand('shell:getprop ro.product.model');
-    return new AndroidDevice(backend, model);
+    const device = new AndroidDevice(android, backend, model.trim());
+    await device._init();
+    return device;
+  }
+
+  async _init() {
+    await this._refreshWebViews();
+    this._pollingWebViews = setInterval(async () => this._refreshWebViews().catch(() => {}), 500);
```

Comment:
I think it's better to employ the Throttler pattern, just in case refreshing web views takes a long time.